docs/helmrelease-guide/tests.md

```diff
@@ -23,6 +23,18 @@ spec:
 When tests are enabled, [resource waiting](release-configuration.md#wait-for-resources-to-be-ready)
 defaults to `true` since this is likely needed for test pre-conditions to be satisfied.
 
+## Rollback release on test failure
+
+RollbackOnFailure will cause a Helm release to be rolled back if the tests fails otherwise it 
+will be left in a released state. rollbackOnFailure is optional and defaults to true 
+
+```yaml
+spec:
+  test:
+    enable: true
+    rollbackOnFailure: true
```

Comment:
There are two scenarios to consider:

1. test fails after install, uninstall?
2. test fails after upgrade, rollback?

I think in most cases one would want the same answer to both of these questions. Also I think anytime a corrective action is taken (uninstall / rollback), the `Released` condition should be marked `false`. So I think a field name which takes all of that into account e.g. `ignore` or `gating` (opposite polarity) may make more sense. Also note the potential for a separate field to account for #452 e.g. `ignoreDelayed`, so that should be considered here.

Reply:
Thanks for your initial feedback @seaneagan  - I have listed below my thoughts:

- At the time, I felt that rollback describe the process of reverting to the last known state (if that is uninstalling as there is no previous state or reverting to a previous release.
- I called it "rollbackOnFailure" as I did not feel 'ignore' gave it enough context. 

Would you be happy with 'ignoreFailures' which would also allow for 'ignoreDelayedFailures'?

p.s this is my first code contribution to helm-operator so I'm new here. It's also the first time I have written Golang so code feedback is welcome. I have another commit adding further code coverage but I am still mastering the go testing framework

Reply:
`ignoreFailures` is perfect :ok_hand: